xxh3.h

```diff
@@ -564,12 +564,12 @@ XXH3_len_4to8_64b(const xxh_u8* input, size_t len, const xxh_u8* secret, XXH64_h
     XXH_ASSERT(input != NULL);
     XXH_ASSERT(secret != NULL);
     XXH_ASSERT(4 <= len && len <= 8);
-    {   xxh_u32 const input_lo = XXH_readLE32(input);
-        xxh_u32 const input_hi = XXH_readLE32(input + len - 4);
-        xxh_u64 const input_64 = input_lo | ((xxh_u64)input_hi << 32);
-        xxh_u64 const keyed = input_64 ^ (XXH_readLE64(secret) + seed);
-        xxh_u64 const mix64 = len + ((keyed ^ (keyed >> 51)) * PRIME32_1);
-        return XXH3_avalanche((mix64 ^ (mix64 >> 47)) * PRIME64_2);
+    {   xxh_u32 const input_1 = XXH_readLE32(input);
+        xxh_u32 const input_2 = XXH_readLE32(input + len - 4);
+        xxh_u64 const input_64 = input_2 | ((xxh_u64)input_1 << 32);
+        xxh_u64 const keyed = input_64 ^ (seed + XXH_readLE64(secret));
+        xxh_u64 const mix64 = len + ((keyed ^ (keyed >> 39)) * PRIME64_2);
+        return XXH3_avalanche(mix64 ^ (mix64 >> 47));
```

Comment:
Removing the multiply improves the strength? Interesting.

Reply:
Nope, it's not stronger. I actually had to select all others parameters very carefully to compensate this operation.
But removing one multiplication from the serial path does improve speed, obviously. That was the main point.

Reply:
Ah, I see. It did have the most multiplies of the three:

| Length | 64-bit muls | 32-bit muls |
|:---|---:|----:|
| 1-3 | 2 | 6 |
| 4-8 | 3 | 8 |
| 9-16 | 2 | 7 |